gxutil/pm.go

```diff
@@ -155,6 +156,73 @@ func (pm *PM) InstallDeps(pkg *Package, location string) error {
 	return pm.installDeps(pkg, location, make(map[string]bool))
 }
 
+// InstallDeps recursively installs all dependencies for the given package
+func (pm *PM) InstallLock(lck *LockFile, location string) error {
+	return pm.installLock(lck, location, make(map[string]bool))
+}
+
+func (pm *PM) installLock(lck *LockFile, location string, complete map[string]bool) error {
```

Comment:
This seems to have pretty much the same functionality as [`fetchDependencies`](https://github.com/whyrusleeping/gx/blob/855ebcf2e2e25d7d55f28f5c35044e8e75264c17/gxutil/pm.go#L754) with the difference that it traverses a `LockFile` instead of a `Package`, could we leverage it? (maybe having a more abstract function that receives a closure that knows how to extract the dependency hashes and install paths?)

Reply:
Yes, absolutely.